scripts/xenadmin-build.ps1

```diff
@@ -0,0 +1,282 @@
+}
+
+function get_locale_id([string]$locale) {
+    switch ($locale) {
+        "ja-jp" { 1041 }
+        "zh-cn" { 2052 }
+        "zh-tw" { 1028 }
+        default { 1033 } #en-us
+    }
+}
+
+$REPO = Get-Item "$PSScriptRoot\.." | Select-Object -ExpandProperty FullName
+$SCRATCH_DIR="$REPO\_scratch"
+$OUTPUT_DIR="$REPO\_output"
+
+mkdir_clean $SCRATCH_DIR
+mkdir_clean $OUTPUT_DIR
+
+. $REPO\scripts\branding.ps1
+$appName =  $BRANDING_BRAND_CONSOLE
+
+#package sources BEFORE applying branding
+
+$gitCommit = git rev-parse HEAD
+git archive --format=zip -o "$SCRATCH_DIR\xenadmin-sources.zip" $gitCommit
+
+Compress-Archive -Path "$SCRATCH_DIR\xenadmin-sources.zip","$REPO\packages\dotnet-packages-sources.zip" `
+    -DestinationPath "$OUTPUT_DIR\$appName-source.zip" -Verbose
+
+#apply branding
+.$REPO\scripts\rebranding.ps1 $buildNumber
+
+Expand-Archive -Path $REPO\packages\XenCenterOVF.zip -DestinationPath $SCRATCH_DIR -Verbose
+build $REPO\XenAdmin.sln
+
+#sign files
+
+if ([System.IO.File]::Exists("$REPO\scripts\sign.ps1")) {
+    . $REPO\scripts\sign.ps1
+
+    $filesToSign = @(
+        "$REPO\XenAdmin\bin\Release\CommandLib.dll",
+        "$REPO\XenAdmin\bin\Release\MSTSCLib.dll",
+        "$REPO\XenAdmin\bin\Release\CoreUtilsLib.dll",
+        "$REPO\XenAdmin\bin\Release\XenModel.dll",
+        "$REPO\XenAdmin\bin\Release\XenOvf.dll",
+        "$REPO\XenAdmin\bin\Release\$appName.exe",
+        "$REPO\xe\bin\Release\xe.exe",
+        "$REPO\XenAdmin\ReportViewer\Microsoft.ReportViewer.Common.dll",
+        "$REPO\XenAdmin\ReportViewer\Microsoft.ReportViewer.ProcessingObjectModel.dll",
+        "$REPO\XenAdmin\ReportViewer\Microsoft.ReportViewer.WinForms.dll",
+        "$REPO\XenAdmin\ReportViewer\Microsoft.ReportViewer.Common.resources.dll",
+        "$REPO\XenAdmin\ReportViewer\Microsoft.ReportViewer.WinForms.resources.dll"
+    )
+
+    foreach ($file in $filesToSign) {
+        sign_artifact $file $appName
+    }
+
+    sign_artifact $REPO\XenAdmin\bin\Release\CookComputing.XmlRpcV2.dll "XML-RPC.NET"
+    sign_artifact $REPO\XenAdmin\bin\Release\Newtonsoft.Json.CH.dll "JSON.NET"
+    sign_artifact $REPO\XenAdmin\bin\Release\log4net.dll "Log4Net"
+    sign_artifact $REPO\XenAdmin\bin\Release\ICSharpCode.SharpZipLib.dll "SharpZipLib"
+    sign_artifact $REPO\XenAdmin\bin\Release\DiscUtils.dll "DiscUtils"
+
+}
+else {
+    Write-Host "Sign script does not exist; skip signing binaries"
+}
+
+#prepare wix
+
+mkdir_clean $SCRATCH_DIR\wixbin
+Expand-Archive -Path $REPO\packages\wix311-binaries.zip -DestinationPath $SCRATCH_DIR\wixbin
+mkdir_clean $SCRATCH_DIR\wixsrc
+Expand-Archive -Path $REPO\packages\wix311-debug.zip -DestinationPath $SCRATCH_DIR\wixsrc
+
+Copy-Item -Recurse $REPO\WixInstaller $SCRATCH_DIR -Verbose
+Copy-Item -Recurse $SCRATCH_DIR\wixsrc\src\ext\UIExtension\wixlib $SCRATCH_DIR\WixInstaller -Verbose
+Copy-Item $SCRATCH_DIR\WixInstaller\wixlib\CustomizeDlg.wxs $SCRATCH_DIR\WixInstaller\wixlib\CustomizeStdDlg.wxs -Verbose
```

Comment:
We should ensure we're escaping square brackets when calling `Copy-Item` in the script. This causes issues when moving files with square brackets in their names, such as `[XenCenter].msi`

Same applies to `Out-File` and `Get-FileHash` calls

Reply:
I went with LiteralPath in the end.